assets/cluster-monitoring-operator/prometheus-rule.yaml

```
@@ -368,6 +368,8 @@ spec:
     rules:
     - expr: sum(openshift_build_total{job="kubernetes-apiservers",phase="Error"})/(sum(openshift_build_total{job="kubernetes-apiservers",phase=~"Failed|Complete|Error"}))
       record: build_error_rate
+    - expr: sum by (strategy) (openshift_build_status_phase_total)
```

one issue with summing counters is that the numbers would be off when a counter resets. It is recommended to rate/increase then sum rather.

```suggestion
    - expr: sum by (strategy) (rate(openshift_build_status_phase_total[5m]))
```

---

Is phase_total a counter?  I thought it was a gauge?  Agree if it's not a gauge this is a bit wierd.

---

from what I can tell from the prometheus target metadata, it is a gauge. @adambkaplan usually the naming convention for Prometheus says that the `_total` suffix is reserved to counters. The correct naming for your metric should be `openshift_build_status_phases` since it's a gauge.
https://prometheus.io/docs/practices/naming/

---

Metric is collected via `openshift-state-metrics` - it's been around since 3.x and is marked STABLE in our GitHub docs. I'm fairly certain this name was chosen before the naming conventions were codified upstream.

https://github.com/openshift/openshift-state-metrics/blob/master/docs/build-metrics.md